File: ar_food_flutter/lib/models/collectible.dart
class Collectible {
  final int? id;
  final String name;
  final String? model3D;
  final String? type;
  final int? xpReward;
  final int? coinReward;
  final double? latitude;
  final double? longitude;
  final bool? available;
  
  Collectible({
    this.id,
    required this.name,
    this.model3D,
    this.type,
    this.xpReward,
    this.coinReward,
    this.latitude,
    this.longitude,
    this.available,
  });
  
  factory Collectible.fromJson(Map<String, dynamic> json) {
    return Collectible(
      id: json['id'],
      name: json['name'],
      model3D: json['model3D'],
      type: json['type'],
      xpReward: json['xpReward'],
      coinReward: json['coinReward'],
      latitude: json['latitude'],
      longitude: json['longitude'],
      available: json['available'],
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'model3D': model3D,
      'type': type,
      'xpReward': xpReward,
      'coinReward': coinReward,
      'latitude': latitude,
      'longitude': longitude,
      'available': available,
    };
  }
}

File: ar-food-backend/src/main/java/com/arfood/entity/Collectible.java
package com.arfood.entity;

import com.arfood.enums.CollectibleType;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "collectibles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Collectible {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    private String model3D;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CollectibleType type;
    
    @Column(name = "xp_reward")
    private Integer xpReward;
    
    @Column(name = "coin_reward")
    private Integer coinReward;
    
    private Double latitude;
    
    private Double longitude;
    
    private Boolean available = true;
    
    @ManyToOne
    @JoinColumn(name = "food_truck_id")
    private FoodTruck foodTruck;
    
    @ManyToMany(mappedBy = "collectibles")
    private Set<Collection> collections = new HashSet<>();
    
    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();
    
    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
    
    public void collect() {
        this.available = false;
    }
}
